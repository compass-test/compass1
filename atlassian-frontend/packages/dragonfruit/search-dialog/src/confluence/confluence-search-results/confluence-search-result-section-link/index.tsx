import {
  SearchResultSectionLink,
  SearchResultSectionLinkProps,
} from '@atlassian/search-dialog';
import React from 'react';
import { useFeatures } from '../../confluence-features';

export default function ConfluenceSearchResultSectionLink(
  props: SearchResultSectionLinkProps,
) {
  const { isMultiSite } = useFeatures();
  if (isMultiSite) {
    return null;
  } else {
    return <SearchResultSectionLink {...props} />;
  }
}
