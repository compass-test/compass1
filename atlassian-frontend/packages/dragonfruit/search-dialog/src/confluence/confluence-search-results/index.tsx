export { default as SearchResults } from './confluence-search-results';
