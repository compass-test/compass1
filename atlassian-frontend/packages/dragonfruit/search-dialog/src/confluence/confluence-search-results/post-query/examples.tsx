import React from 'react';
import { IntlProvider } from 'react-intl';
import styled from 'styled-components';
import { ConfluencePostQuery } from './confluence-post-query';
import {
  createPageBlogAttachmentResults,
  createPeopleResults,
  createSpaceResponse,
} from '../../../__tests__/__fixtures__/mock-search-results';
import { action } from '@storybook/addon-actions';
import { ScreenType } from '../../../common/analytics/events';
import { SearchSessionProvider } from '../../../common/search-session-provider';

const defaultProps = {
  items: {
    isLoading: false,
    results: createPageBlogAttachmentResults(3),
  },
  spaces: {
    isLoading: false,
    results: createSpaceResponse(3),
  },
  people: {
    isLoading: false,
    results: createPeopleResults(3),
  },
  query: 'abc',
  enabledFilters: [],
  searchResultSelected: action('search result selected'),
  advancedSearchSelected: action('advanced search clicked'),
  showMoreClicked: action('show more clicked'),
  showAllClicked: action('show all clicked'),
  searchSessionId: 'fakeSearchSessionId',
  screenType: 'postQuerySearchResults' as ScreenType,
  isLoading: false,
  isCollapsed: false,
  services: {
    isLoading: false,
    cursor: null,
    results: [],
  },
  libsAppsMore: {
    isLoading: false,
    cursor: null,
    results: [],
  },
  peopleTeams: {
    isLoading: false,
    results: [],
  },
  fetchMoreFn: () => void 0,
  teamNames: {},
};

const emptyProps = {
  ...defaultProps,
  items: {
    isLoading: false,
    results: {
      items: [],
      totalSize: 0,
      timings: 0,
    },
  },
  spaces: {
    isLoading: false,
    results: {
      items: [],
      timings: 0,
    },
  },
  people: {
    isLoading: false,
    results: {
      items: [],
      timings: 0,
    },
  },
  services: {
    isLoading: false,
    cursor: null,
    results: [],
  },
  libsAppsMore: {
    isLoading: false,
    cursor: null,
    results: [],
  },
  peopleTeams: {
    isLoading: false,
    results: [],
  },
  fetchMoreFn: () => void 0,
  teamNames: {},
};

const Container = styled.div`
  height: calc(100vh - 160px);
  width: 780px;
  border: 1px solid black;
  padding: 8px 0;
`;

export const DefaultPostQuery = () => <ConfluencePostQuery {...defaultProps} />;

export const PostQueryCollapsed = () => (
  <ConfluencePostQuery {...defaultProps} isCollapsed />
);

export const NoResultsPostQuery = () => <ConfluencePostQuery {...emptyProps} />;

export const NoItemPostQuery = () => (
  <ConfluencePostQuery {...defaultProps} items={emptyProps.items} />
);

export const NoSpacePostQuery = () => (
  <ConfluencePostQuery {...defaultProps} spaces={emptyProps.spaces} />
);

export const NoPeoplePostQuery = () => (
  <ConfluencePostQuery {...defaultProps} people={emptyProps.people} />
);

export const ItemsPostQuery25 = () => (
  <ConfluencePostQuery
    {...defaultProps}
    items={{
      isLoading: false,
      results: {
        ...createPageBlogAttachmentResults(25),
        totalSize: 25,
      },
    }}
  />
);

export const ItemsPostQuery150 = () => (
  <ConfluencePostQuery
    {...defaultProps}
    items={{
      isLoading: false,
      results: {
        ...createPageBlogAttachmentResults(30),
        totalSize: 150,
      },
    }}
  />
);

export const PostQueryRenderingIntermediateResultsFasterSearch = () => (
  <ConfluencePostQuery {...defaultProps} screenType="cachedResults" />
);

export default {
  title: 'Confluence Search Dialog/Confluence Post Query',
  decorators: [
    (story: () => React.ElementType) => (
      <Container>
        <SearchSessionProvider sessionKey="story_session_key">
          <IntlProvider locale="en">{story()}</IntlProvider>
        </SearchSessionProvider>
      </Container>
    ),
  ],
};
