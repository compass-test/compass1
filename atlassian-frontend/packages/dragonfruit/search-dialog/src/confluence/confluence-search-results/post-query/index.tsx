export { default as ConfluencePostQuery } from './confluence-post-query';
