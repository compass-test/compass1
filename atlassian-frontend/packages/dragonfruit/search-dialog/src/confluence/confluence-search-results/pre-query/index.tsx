export { default as ConfluencePreQuery } from './confluence-pre-query';
