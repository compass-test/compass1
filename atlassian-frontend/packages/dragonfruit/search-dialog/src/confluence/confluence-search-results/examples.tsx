import React from 'react';
import { IntlProvider } from 'react-intl';
import styled from 'styled-components';
import { action } from '@storybook/addon-actions';
import { SearchResults } from './confluence-search-results';
import {
  createPageBlogAttachmentResults,
  createPeopleResults,
  createSpaceResponse,
} from '../../__tests__/__fixtures__/mock-search-results';
import { SearchSessionProvider } from '../../common/search-session-provider';
import { FilterContextProvider } from '../filter-context';
import { MockConfluenceClientsProvider } from '../../__tests__/__fixtures__/mock-confluence-clients-provider';
import { ABTestProvider, DEFAULT_AB_TEST } from '../../common/ab-test-provider';
import { SharedClient } from '../../common/clients';
import { Products, ProductProvider } from '../../common/product-context';

const LoadingResultState = {
  isLoading: true,
  results: null,
};

const defaultProps = {
  query: 'abc',
  advancedSearchUrl: 'some/advanced/search/url?q=abc',
  enabledFilters: [],
  onFiltersCleared: action('filters cleared'),
  items: {
    isLoading: false,
    results: createPageBlogAttachmentResults(3),
  },
  spaces: {
    isLoading: false,
    results: createSpaceResponse(3),
  },
  people: {
    isLoading: false,
    results: createPeopleResults(3),
  },
  services: {
    isLoading: false,
    cursor: null,
    results: null,
  },
  libsAppsMore: {
    isLoading: false,
    cursor: null,
    results: null,
  },
  peopleTeams: {
    isLoading: false,
    results: null,
  },
  showFilters: true,
  isBelowTakeoverWidth: false,
  fetchMoreFn: () => void 0,
  teamNames: {},
  recentlyViewedComponents: { loading: false, components: [] },
  recentlyViewedTeams: { loading: false, teams: [] },
};

const emptyProps = {
  ...defaultProps,
  items: {
    isLoading: false,
    results: createPageBlogAttachmentResults(0),
  },
  spaces: {
    isLoading: false,
    results: createSpaceResponse(0),
  },
  people: {
    isLoading: false,
    results: createPeopleResults(0),
  },
  services: {
    isLoading: false,
    cursor: null,
    results: null,
  },
  libsAppsMore: {
    isLoading: false,
    cursor: null,
    results: null,
  },
  peopleTeams: {
    isLoading: false,
    results: null,
  },
  showFilters: true,
  teamNames: {},
};

const loadingProps = {
  ...defaultProps,
  items: LoadingResultState,
  spaces: LoadingResultState,
  people: LoadingResultState,
};

const Container = styled.div`
  width: 780px;
  border: 1px solid black;
`;

const TinyContainer = styled.div`
  width: 780px;
  height: 200px;
`;

const MockSharedClient: SharedClient = {
  getAbTestData: () => Promise.resolve(DEFAULT_AB_TEST),
  getProductPermissions: async () => [],
};

export const PreDefault = () => <SearchResults {...defaultProps} isPreQuery />;

export const PreLoading = () => <SearchResults {...loadingProps} isPreQuery />;

export const PreNoResults = () => <SearchResults {...emptyProps} isPreQuery />;

export const PreTinyContainer = () => (
  <TinyContainer>
    <SearchResults {...defaultProps} isPreQuery />
  </TinyContainer>
);

export const PostDefault = () => (
  <SearchResults {...defaultProps} isPreQuery={false} />
);

export const PostLoading = () => (
  <SearchResults {...loadingProps} isPreQuery={false} />
);

export const PostNoResults = () => (
  <SearchResults {...emptyProps} isPreQuery={false} />
);

export const PostTinyContainer = () => (
  <TinyContainer>
    <SearchResults {...defaultProps} isPreQuery={false} />
  </TinyContainer>
);

export const PostAndFilterDefault = () => (
  <SearchResults
    {...defaultProps}
    isPreQuery={false}
    showFilters
    isBelowTakeoverWidth={false}
  />
);

export const PostAndFilterLoading = () => (
  <SearchResults
    {...loadingProps}
    isPreQuery={false}
    showFilters
    isBelowTakeoverWidth={false}
  />
);

export const PostAndFilterNoResults = () => (
  <SearchResults
    {...emptyProps}
    isPreQuery={false}
    showFilters
    isBelowTakeoverWidth={false}
  />
);

export const PostAndFilterTinyContainer = () => (
  <TinyContainer>
    <SearchResults
      {...defaultProps}
      isPreQuery={false}
      showFilters
      isBelowTakeoverWidth={false}
    />
  </TinyContainer>
);

export const PostAndFilterFiltersWithFewResults = () => (
  <SearchResults
    {...emptyProps}
    isPreQuery={false}
    showFilters
    isBelowTakeoverWidth={false}
  />
);

export const LinkComponent = () => (
  <SearchResults
    {...defaultProps}
    isPreQuery={false}
    /* eslint-disable-next-line */
    linkComponent={(props: any) => (
      <a {...props} href={undefined} onClick={action('Link clicked')}>
        Click Me
      </a>
    )}
  />
);

export default {
  title: 'Confluence Search Dialog/Confluence Search Results',
  decorators: [
    (story: () => React.ElementType) => (
      <Container>
        <MockConfluenceClientsProvider mode="normal">
          <SearchSessionProvider sessionKey="story_session_key">
            <ABTestProvider searchClient={MockSharedClient}>
              <ProductProvider products={[Products.confluence]}>
                <IntlProvider locale="en">
                  <FilterContextProvider>{story()}</FilterContextProvider>
                </IntlProvider>
              </ProductProvider>
            </ABTestProvider>
          </SearchSessionProvider>
        </MockConfluenceClientsProvider>
      </Container>
    ),
  ],
};
