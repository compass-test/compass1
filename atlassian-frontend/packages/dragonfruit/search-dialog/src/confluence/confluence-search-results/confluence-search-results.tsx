import React from 'react';
import { LinkComponent } from '@atlassian/search-dialog';
import { ConfPeopleResults } from '../clients/response-types';
import { ConfluencePreQuery } from './pre-query';
import { ConfluencePostQuery } from './post-query';
import { LoadingSpinner } from '../../common/loading-spinner';
import { usePrimarySiteConfluenceAdvancedSearchUrlFactory } from '../confluence-utils/confluence-url-utils';
import NoResults from './confluence-search-no-results';
import { CompassSearchComponentNodeFragment } from '@atlassian/dragonfruit-graphql';
import { TeamDetails } from '@atlassian/dragonfruit-rest';
import { User } from '@atlaskit/user-picker';
import {
  RecentComponentsState,
  RecentTeamsState,
} from '@atlassian/compass-search-cache';

interface Props {
  query: string;
  isPreQuery: boolean;
  linkComponent?: LinkComponent;
  formatDate?: (lastModified: string) => JSX.Element;
  services: {
    isLoading: boolean;
    cursor: string | null;
    results: CompassSearchComponentNodeFragment[] | null;
  };
  libsAppsMore: {
    isLoading: boolean;
    cursor: string | null;
    results: CompassSearchComponentNodeFragment[] | null;
  };
  peopleTeams: {
    isLoading: boolean;
    results: (TeamDetails | User)[] | null;
  };
  people: {
    isLoading: boolean;
    results: ConfPeopleResults | null;
  };
  isBelowTakeoverWidth: boolean;
  showFilters?: boolean;
  fetchMoreFn: (typeResult: string, cursor: string | null) => void;
  teamNames: { [key: string]: TeamDetails | null };
  recentlyViewedComponents: RecentComponentsState;
  recentlyViewedTeams: RecentTeamsState;
}

interface ContextProps {
  advancedSearchUrl: string;
}

export enum ResultState {
  PreQueryLoading,
  PostQueryLoading,

  PreQueryResults,
  PostQueryResults,
}

export const getCurrentState = (props: Props) => {
  const {
    services,
    libsAppsMore,
    peopleTeams,
    isPreQuery,
    recentlyViewedComponents,
  } = props;
  const isLoading =
    services.isLoading || libsAppsMore.isLoading || peopleTeams.isLoading;

  if (isPreQuery) {
    return recentlyViewedComponents.loading
      ? ResultState.PreQueryLoading
      : ResultState.PreQueryResults;
  }

  return isLoading
    ? ResultState.PostQueryLoading
    : ResultState.PostQueryResults;
};

export class SearchResults extends React.Component<Props & ContextProps> {
  static getFetchedResultsCount(props: Props) {
    const { services, libsAppsMore, peopleTeams } = props;

    const serviceLength = (services.results && services.results.length) || 0;
    const libsAppsMoreLength =
      (libsAppsMore.results && libsAppsMore.results.length) || 0;
    const peopleTeamsLength =
      (peopleTeams.results && peopleTeams.results.length) || 0;

    return {
      serviceLength,
      libsAppsMoreLength,
      peopleTeamsLength,
      totalCount: serviceLength + libsAppsMoreLength + peopleTeamsLength,
    };
  }

  static hasResults(props: Props) {
    return SearchResults.getFetchedResultsCount(props).totalCount > 0;
  }

  render() {
    const {
      query,
      services,
      libsAppsMore,
      peopleTeams,
      people,
      linkComponent,
      formatDate,
      isBelowTakeoverWidth,
      showFilters = false,
      advancedSearchUrl,
      fetchMoreFn,
      teamNames,
      recentlyViewedComponents,
      recentlyViewedTeams,
    } = this.props;
    const currentState = getCurrentState(this.props);
    const hasResults = SearchResults.hasResults(this.props);
    let results;

    switch (currentState) {
      case ResultState.PreQueryLoading: {
        results = <LoadingSpinner />;
        break;
      }
      case ResultState.PostQueryLoading: {
        results = (
          <>
            {hasResults ? (
              <ConfluencePostQuery
                query={query}
                services={services}
                libsAppsMore={libsAppsMore}
                peopleTeams={peopleTeams}
                // TODO: remove these
                items={{ isLoading: false, results: null }}
                spaces={{ isLoading: false, results: null }}
                people={people}
                fetchMoreFn={fetchMoreFn}
                teamNames={teamNames}
                linkComponent={linkComponent}
                formatDate={formatDate}
                isLoading
                screenType="cachedResults"
                isCollapsed={isBelowTakeoverWidth || showFilters}
              />
            ) : null}
            <LoadingSpinner />
          </>
        );
        break;
      }
      case ResultState.PreQueryResults: {
        results = (
          <>
            <ConfluencePreQuery
              components={recentlyViewedComponents.components}
              teams={recentlyViewedTeams.teams}
              linkComponent={linkComponent}
              isLoading={false}
              isCollapsed={isBelowTakeoverWidth}
              screenType={'cachedResults'}
            />
          </>
        );
        break;
      }
      case ResultState.PostQueryResults: {
        results = (
          <>
            {hasResults ? (
              <ConfluencePostQuery
                query={query}
                services={services}
                libsAppsMore={libsAppsMore}
                peopleTeams={peopleTeams}
                items={{ isLoading: false, results: null }}
                spaces={{ isLoading: false, results: null }}
                people={people}
                fetchMoreFn={fetchMoreFn}
                teamNames={teamNames}
                linkComponent={linkComponent}
                isLoading={
                  services.isLoading ||
                  libsAppsMore.isLoading ||
                  peopleTeams.isLoading
                }
                formatDate={formatDate}
                screenType="postQuerySearchResults"
                isCollapsed={isBelowTakeoverWidth || showFilters}
              />
            ) : (
              <NoResults
                linkComponent={linkComponent}
                advancedSearchUrl={advancedSearchUrl}
              />
            )}
          </>
        );
        break;
      }
      default: {
        // This is to ensure the switch is exhaustive
        const error: never = currentState;
        throw new Error(error);
      }
    }
    return results;
  }
}

export default (props: Props) => {
  const advancedSearchUrl = usePrimarySiteConfluenceAdvancedSearchUrlFactory()(
    props.query,
  );
  return <SearchResults {...props} advancedSearchUrl={advancedSearchUrl} />;
};
