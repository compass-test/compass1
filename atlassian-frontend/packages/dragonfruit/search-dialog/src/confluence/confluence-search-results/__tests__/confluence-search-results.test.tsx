import React from 'react';
import { shallow } from 'enzyme';
import { enableDeterministicResponses } from '../../../__tests__/__fixtures__/mock-search-results';
import { SearchResults } from '../confluence-search-results';
import { ConfluencePreQuery } from '../pre-query';
import { ConfluencePostQuery } from '../post-query';
import { LoadingSpinner } from '../../../common/loading-spinner';
import { DefaultFeatures } from '../../confluence-features';
import {
  CompassComponentType,
  CompassSearchComponentResult,
} from '@atlassian/dragonfruit-graphql';

jest.mock('../pre-query');
jest.mock('../post-query');
jest.mock('../../../common/loading-spinner', () => ({
  LoadingSpinner: () => 'div',
}));
jest.mock('@atlaskit/width-detector', () => ({
  WidthObserver: () => <div />,
}));
jest.mock('../../filter');

describe('<SearchResults />', () => {
  let fetchMoreFn: any;

  beforeEach(() => {
    fetchMoreFn = jest.fn();
  });

  const names = {
    '1eb19c9d-e4c7-45c8-aec3-8a2d7cbe5515': {
      displayName: 'this is a team',
    } as any,
    '15c3f236-8339-43e0-9049-2fd03e808d9d': {
      displayName: 'another team',
    } as any,
    '5a740f67-ee17-4558-b567-60116bcd266d': {
      displayName: 'yet another team',
    } as any,
  };

  const createResults = () => ({
    services: {
      isLoading: false,
      results: [
        {
          component: {
            id: 'id-123',
            name: 'name-123',
            description: 'description-text-Lorem ipsum',
            ownerId: '1eb19c9d-e4c7-45c8-aec3-8a2d7cbe5515',
          },
          link: 'https://link',
        } as CompassSearchComponentResult,
      ],
      cursor: null,
    },
    libsAppsMore: {
      isLoading: false,
      results: [],
      cursor: null,
    },
    peopleTeams: {
      isLoading: false,
      results: [],
    },
    people: {
      isLoading: false,
      results: null,
    },
    fetchMoreFn,
    teamNames: names,
  });

  const LoadingResultState = {
    isLoading: true,
    results: null,
  };

  const loadingProps = {
    services: { ...LoadingResultState, cursor: null },
    libsAppsMore: { ...LoadingResultState, cursor: null },
    peopleTeams: LoadingResultState,
    people: LoadingResultState,
  };

  const baseProps = {
    query: 'query',
    advancedSearchUrl: 'some/advanced/search/url?q=query',
    enabledFilters: [],
    onSearchResultsShown: jest.fn(),
    searchSessionId: 'testSearchSeassionId',
    features: DefaultFeatures,
    isBelowTakeoverWidth: false,
    recentlyViewedComponents: { loading: false, components: [] },
    recentlyViewedTeams: { loading: false, teams: [] },
  };

  beforeAll(() => {
    enableDeterministicResponses();
  });

  it('render with results matches snapshot for pre query', () => {
    const wrapper = shallow(
      <SearchResults {...baseProps} {...createResults()} isPreQuery />,
    );

    expect(wrapper.find(ConfluencePreQuery).exists()).toBe(true);
    expect(wrapper.find(ConfluencePostQuery).exists()).toBe(false);
    expect(wrapper).toMatchSnapshot();
  });

  it('render with results matches snapshot for pre query with components and teams', () => {
    const wrapper = shallow(
      <SearchResults
        {...baseProps}
        {...createResults()}
        isPreQuery
        recentlyViewedComponents={{
          loading: false,
          components: [
            {
              id: 'id1',
              name: 'component-1',
              type: CompassComponentType.SERVICE,
            },
          ],
        }}
        recentlyViewedTeams={{
          loading: false,
          teams: [
            {
              id: 'team1',
              displayName: 'Lodestone',
              smallAvatarImageUrl: 'https://atlassian.net/test.svg',
            },
          ],
        }}
      />,
    );

    expect(wrapper.find(ConfluencePreQuery).exists()).toBe(true);
    expect(wrapper.find(ConfluencePostQuery).exists()).toBe(false);
    expect(wrapper).toMatchSnapshot();
  });

  it('render with results matches snapshot for post query', () => {
    const wrapper = shallow(
      <SearchResults {...baseProps} {...createResults()} isPreQuery={false} />,
    );

    expect(wrapper.find(ConfluencePreQuery).exists()).toBe(false);
    expect(wrapper.find(ConfluencePostQuery).exists()).toBe(true);
    expect(wrapper.find(ConfluencePostQuery).prop('screenType')).toBe(
      'postQuerySearchResults',
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('render with results matches snapshot for pre query loading', () => {
    const wrapper = shallow(
      <SearchResults
        {...baseProps}
        fetchMoreFn={fetchMoreFn}
        teamNames={{}}
        {...loadingProps}
        isPreQuery
        recentlyViewedComponents={{ loading: true, components: [] }}
      />,
    );

    expect(wrapper.find(ConfluencePreQuery).exists()).toBe(false);
    expect(wrapper.find(ConfluencePostQuery).exists()).toBe(false);
    expect(wrapper.find(LoadingSpinner).exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });

  describe('with no results', () => {
    let wrapper: any;
    beforeEach(() => {
      baseProps.onSearchResultsShown = jest.fn();
      wrapper = shallow(
        <SearchResults
          {...baseProps}
          fetchMoreFn={fetchMoreFn}
          teamNames={{}}
          {...loadingProps}
          isPreQuery={false}
        />,
      );
    });

    it('render and matches snapshot for post query loading', () => {
      expect(wrapper.find(ConfluencePreQuery).exists()).toBe(false);
      expect(wrapper.find(ConfluencePostQuery).exists()).toBe(false);
      expect(wrapper.find(LoadingSpinner).exists()).toBe(true);
      expect(wrapper).toMatchSnapshot();
    });
  });

  it('render with intermediate results matches snapshot for post query loading', () => {
    const wrapper = shallow(
      <SearchResults
        {...baseProps}
        {...loadingProps}
        services={{
          isLoading: false,
          cursor: null,
          results: [
            {
              component: {
                id: 'id-123',
                name: 'name-123',
                description: 'description-text-Lorem ipsum',
                ownerId: '123',
              },
              link: 'https://link',
            } as CompassSearchComponentResult,
          ],
        }}
        libsAppsMore={{
          isLoading: false,
          cursor: null,
          results: [],
        }}
        isPreQuery={false}
        fetchMoreFn={fetchMoreFn}
        teamNames={{}}
      />,
    );

    expect(wrapper.find(ConfluencePreQuery).exists()).toBe(false);

    // We expect the post query screen to show here along with the spinner
    expect(wrapper.find(ConfluencePostQuery).exists()).toBe(true);
    expect(wrapper.find(ConfluencePostQuery).prop('screenType')).toBe(
      'cachedResults',
    );
    expect(wrapper.find(LoadingSpinner).exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
});
