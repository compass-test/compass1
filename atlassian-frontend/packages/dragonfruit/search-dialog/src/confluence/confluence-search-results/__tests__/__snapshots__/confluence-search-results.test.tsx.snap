// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResults /> render with intermediate results matches snapshot for post query loading 1`] = `
<Fragment>
  <WithAnalyticsEvents(WrappedComponent)
    fetchMoreFn={[MockFunction]}
    isCollapsed={false}
    isLoading={true}
    items={
      Object {
        "isLoading": false,
        "results": null,
      }
    }
    libsAppsMore={
      Object {
        "cursor": null,
        "isLoading": false,
        "results": Array [],
      }
    }
    people={
      Object {
        "isLoading": true,
        "results": null,
      }
    }
    peopleTeams={
      Object {
        "isLoading": true,
        "results": null,
      }
    }
    query="query"
    screenType="cachedResults"
    services={
      Object {
        "cursor": null,
        "isLoading": false,
        "results": Array [
          Object {
            "component": Object {
              "description": "description-text-Lorem ipsum",
              "id": "id-123",
              "name": "name-123",
              "ownerId": "123",
            },
            "link": "https://link",
          },
        ],
      }
    }
    spaces={
      Object {
        "isLoading": false,
        "results": null,
      }
    }
    teamNames={Object {}}
  />
  <LoadingSpinner />
</Fragment>
`;

exports[`<SearchResults /> render with results matches snapshot for post query 1`] = `
<Fragment>
  <WithAnalyticsEvents(WrappedComponent)
    fetchMoreFn={[MockFunction]}
    isCollapsed={false}
    isLoading={false}
    items={
      Object {
        "isLoading": false,
        "results": null,
      }
    }
    libsAppsMore={
      Object {
        "cursor": null,
        "isLoading": false,
        "results": Array [],
      }
    }
    people={
      Object {
        "isLoading": false,
        "results": null,
      }
    }
    peopleTeams={
      Object {
        "isLoading": false,
        "results": Array [],
      }
    }
    query="query"
    screenType="postQuerySearchResults"
    services={
      Object {
        "cursor": null,
        "isLoading": false,
        "results": Array [
          Object {
            "component": Object {
              "description": "description-text-Lorem ipsum",
              "id": "id-123",
              "name": "name-123",
              "ownerId": "1eb19c9d-e4c7-45c8-aec3-8a2d7cbe5515",
            },
            "link": "https://link",
          },
        ],
      }
    }
    spaces={
      Object {
        "isLoading": false,
        "results": null,
      }
    }
    teamNames={
      Object {
        "15c3f236-8339-43e0-9049-2fd03e808d9d": Object {
          "displayName": "another team",
        },
        "1eb19c9d-e4c7-45c8-aec3-8a2d7cbe5515": Object {
          "displayName": "this is a team",
        },
        "5a740f67-ee17-4558-b567-60116bcd266d": Object {
          "displayName": "yet another team",
        },
      }
    }
  />
</Fragment>
`;

exports[`<SearchResults /> render with results matches snapshot for pre query 1`] = `
<Fragment>
  <WithAnalyticsEvents(WrappedComponent)
    components={Array []}
    isCollapsed={false}
    isLoading={false}
    screenType="cachedResults"
    teams={Array []}
  />
</Fragment>
`;

exports[`<SearchResults /> render with results matches snapshot for pre query loading 1`] = `<LoadingSpinner />`;

exports[`<SearchResults /> render with results matches snapshot for pre query with components and teams 1`] = `
<Fragment>
  <WithAnalyticsEvents(WrappedComponent)
    components={
      Array [
        Object {
          "id": "id1",
          "name": "component-1",
          "type": "SERVICE",
        },
      ]
    }
    isCollapsed={false}
    isLoading={false}
    screenType="cachedResults"
    teams={
      Array [
        Object {
          "displayName": "Lodestone",
          "id": "team1",
          "smallAvatarImageUrl": "https://atlassian.net/test.svg",
        },
      ]
    }
  />
</Fragment>
`;

exports[`<SearchResults /> with no results render and matches snapshot for post query loading 1`] = `
<Fragment>
  <LoadingSpinner />
</Fragment>
`;
